python/sglang/srt/compilation/piecewise_context_manager.py

```diff
@@ -48,13 +53,17 @@ def get_forward_context() -> Optional[ForwardContext]:
 
 @contextmanager
 def set_forward_context(
-    forward_batch: ForwardBatch, attention_layers: List[Any], quant_config: Any
+    forward_batch: ForwardBatch,
+    attention_layers: List[Any],
+    quant_config: Any,
+    moe_layers: List[Any],
 ):
     global _forward_context
     _forward_context = ForwardContext()
     _forward_context.set_forward_batch(forward_batch)
     _forward_context.set_attention_layers(attention_layers)
     _forward_context.set_quant_config(quant_config)
+    _forward_context.set_moe_layers(moe_layers)
```

Comment:
These setters are redundant. Can you just put them in the `__init__` of `ForwardContext`?